### PR TITLE
Upgrade formatting, readability, consistency in configure.ac.

### PR DESCRIPTION
**Pull request by randy-waterhouse**
Helps tidy ./configure --help readability and consistency also. Needs testing on OSX, windows builds.

### CODE REVIEW

**Comment by jgarzik** [comment]
ut ACK -- seems like @theuni trivial tree material

**Comment by randy-waterhouse** [comment]
Adding some more as I work through.
@jgarzik should just be trivial if I steer clear of that gray line between formatting and functionality.

File: configure.ac [hunk 1]
```diff
@@ -52,178 +53,186 @@ dnl Libtool init checks.
...
-     AC_CHECK_LIB([advapi32],      [main],, AC_MSG_ERROR(lib missing))
-     AC_CHECK_LIB([ws2_32],      [main],, AC_MSG_ERROR(lib missing))
-     AC_CHECK_LIB([mswsock],      [main],, AC_MSG_ERROR(lib missing))
-     AC_CHECK_LIB([shlwapi],      [main],, AC_MSG_ERROR(lib missing))
-     AC_CHECK_LIB([iphlpapi],      [main],, AC_MSG_ERROR(lib missing))
-     AC_CHECK_LIB([crypt32],      [main],, AC_MSG_ERROR(lib missing))
-
-     # -static is interpreted by libtool, where it has a different meaning.
-     # In libtool-speak, it's -all-static.
+     AC_CHECK_LIB([mingwthrd],[main],[], AC_MSG_ERROR([lib missing]))
```
Comment (laanwj) [inline configure.ac]: Although it adds a bit of readability, 'pad with spaces' alignment has the disadvantage that a change on one line makes it necessary to change every line in the table, which makes reviewing changes harder, and can introduce merge conflicts. So I'm generally not in favor of it.

**Comment by laanwj** [comment]
I'll leave this up to @theuni.

I wonder if there is something akin to clang-format for these files. I'm not against increasing readability, but it is very subjective. At least try to do this all at once, so that we don't break current pulls affecting `configure.ac` multiple times.

**Comment by theuni** [comment]
Sorry for the slow response here, I've been trying to get to the bottom of the AWS block file corruption. I'll review tomorrow.

**Comment by theuni** [comment]
@laanwj My thoughts exactly. These are all good cleanups, but without any real formatting rules this is basically a never-ending syntax ping-pong.

@randy-waterhouse I'm not thrilled about all the whitespace/formatting changes because as I see it one of two things will happen:
1. These will go in, but the next set of changes won't necessarily match.
2. These go in, then we start policing nits in the future (see header includes as an example there).

My preference would be to take the stuff that is more technically correct (# -> dnl and adding missing [ ] for example), and leave the stuff that's just internal formatting alone.

Re-ordering the options makes sense though, I like it when other programs have theirs organized.

Either way, let's take this over to the trivial tree to discuss, to avoid burning c-i cycles here. Please PR to trivial-next in https://github.com/theuni/bitcoin.

**Comment by randy-waterhouse** [comment]
I agree, it is too trivial to waste our time on a back and forward about formatting style preferences.
You seem to know what you want and you can take from my mods whatever you like. 

The file does need cleaning up though it is in a bit of a state.

**Pull request closed without merging**